routing/router.go

```diff
@@ -1545,69 +1545,119 @@ func getEdgeUnifiers(source route.Vertex, hops []route.Vertex,
 // determines the amount that should be sent to fulfill min HTLC requirements.
 // The minimal sender amount can be searched for by activating useMinAmt.
 func senderAmtBackwardPass(unifiers []*edgeUnifier, useMinAmt bool,
-	runningAmt lnwire.MilliSatoshi,
+	receiverAmt lnwire.MilliSatoshi,
 	bandwidthHints bandwidthHints) ([]*unifiedEdge, lnwire.MilliSatoshi,
 	error) {
 
 	var unifiedEdges = make([]*unifiedEdge, len(unifiers))
 
-	// Traverse hops backwards to accumulate fees in the running amounts.
-	for i := len(unifiers) - 1; i >= 0; i-- {
-		localChan := i == 0
-		edgeUnifier := unifiers[i]
+	// We traverse the route backwards and handle the last hop separately.
+	edgeUnifier := unifiers[len(unifiers)-1]
```

Comment:
Do we need a bounds check here or are we certain there's always at least one edge?

Reply:
nice, I added a check here and on the RPC level (new commit)